### PR TITLE
PSP-3011 unit tests

### PR DESCRIPTION
**Pull request by asanchezr**

### CODE REVIEW

File: backend/dal/Repositories/PropertyRepository.cs [hunk 1]
```diff
@@ -136,8 +136,7 @@ public PimsProperty Get(long id)
                     .ThenInclude(l => l.Lease)
                     .ThenInclude(l => l.PimsLeaseTenants)
                     .ThenInclude(l => l.Organization)
-                .Where(p => p.PropertyId == id)
-                .FirstOrDefault() ?? throw new KeyNotFoundException();
+                .FirstOrDefault(p => p.PropertyId == id) ?? throw new KeyNotFoundException();
```
Comment (asanchezr) [inline backend/dal/Repositories/PropertyRepository.cs:139]: stylecop suggestion

File: backend/entities/Property.cs [hunk 2]
```diff
@@ -60,6 +60,8 @@ public PimsProperty(int pid, PimsPropertyType type, PimsPropertyClassificationTy
             this.PropertyDataSourceTypeCodeNavigation = dataSource ?? throw new ArgumentNullException(nameof(dataSource));
             this.PropertyDataSourceTypeCode = dataSource.Id;
             this.PropertyDataSourceEffectiveDate = dataSourceEffectiveDate;
+            this.PropertyStatusTypeCodeNavigation = status ?? throw new ArgumentNullException(nameof(status));
+            this.PropertyStatusTypeCode = status.Id;
```
Comment (asanchezr) [inline backend/entities/Property.cs:64]: Required relationship was missing on unit tests

File: backend/api/Areas/Property/Controllers/PropertyController.cs [hunk 3]
```diff
@@ -112,15 +112,15 @@ public IActionResult GetConceptPropertyWithPid(string pid)
...
+        /// Update the specified property, and attached properties.
         /// </summary>
         /// <returns></returns>
         [HttpPut("concept/{pid}")]
         [HasPermission(Permissions.PropertyEdit)]
         [Produces("application/json")]
         [ProducesResponseType(typeof(Pims.Api.Models.Concepts.PropertyModel), 200)]
         [SwaggerOperation(Tags = new[] { "property" })]
-        public IActionResult UpdatePerson([FromBody] Pims.Api.Models.Concepts.PropertyModel propertyModel)
+        public IActionResult UpdateConceptProperty([FromBody] Pims.Api.Models.Concepts.PropertyModel propertyModel)
```
Comment (FuriousLlama) [inline backend/api/Areas/Property/Controllers/PropertyController.cs:123]: Why the change of name?

File: backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        public void GetPage_Properties(PropertyFilter filter, int expectedCount)
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyView);
+
+            using var init = helper.InitializeDatabase(user);
+
+            init.CreateProperty(2);
+            init.CreateProperty(3, pin: 111);
```
Comment (FuriousLlama) [inline backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs:86]: I think creating a property and passing os more readable than just calling CreateProperty

File: backend/api/Areas/Property/Controllers/PropertyController.cs (hunk 3 above)
Comment (asanchezr) [reply backend/api/Areas/Property/Controllers/PropertyController.cs:123]: It was a copy/paste mistake from previous PR - this is the property controller, not the person controller

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        #region Tests
+        #region GetById
+        [Fact]
+        public void GetById_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyView);
+
+            var property = EntityHelper.CreateProperty(1);
```
Comment (FuriousLlama) [inline backend/tests/unit/dal/Services/PropertyServiceTest.cs:31]: it does not look like the property is being used. Is it necessary to be created?

Comment (asanchezr) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:31]: Look at the next line - it is adding it to the in-memory DB

```cs
helper.CreatePimsContext(user, true).AddAndSaveChanges(property);
```

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(property);
```
Comment (FuriousLlama) [inline backend/tests/unit/dal/Services/PropertyServiceTest.cs:32]: if we are testing the service, is the context necessary?

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs (hunk 5 above)
Comment (asanchezr) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:31]: We need to add it to be able to find it by ID - the whole purpose of this test

File: backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+            // Assert
+            Assert.Throws<KeyNotFoundException>(() => repository.Update(property));
+        }
+
+        [Fact]
+        public void Update_Property_ThrowIfNull()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.PropertyView);
```
Comment (FuriousLlama) [inline backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs:196]: is the user relevant for repository tests?

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs (hunk 6 above)
Comment (asanchezr) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:32]: I guess we might not need it but I followed other existing tests that do that. Wouldn't want to go a different way and find that it doesn't work. In any case, do you see harm in that?

File: backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs (hunk 4 above)
Comment (asanchezr) [reply backend/tests/unit/dal/Repositories/PropertyRepositoryTest.cs:86]: This is an existing test - it appears as new because I renamed this file from PropertyServiceTest -> PropertyRepositoryTest

I'd lean to not changing it for now as it has been there before - unless there is something fundamentally wrong with it

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs (hunk 6 above)
Comment (FuriousLlama) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:32]: Is just that any code that does not add value should be removed if possible.

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs (hunk 5 above)
Comment (FuriousLlama) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:31]: Looks like the mock on line 36 is not checking the ID passed. Line 42 is not checking that the property passed is being returned (In this case wont be returned given the repository mocked function is not using the created context). So from that I gather that the property is not being used for the purpose of the test

**Review by FuriousLlama — APPROVED** [review]
It would be a good idea to revisit our testing practices

File: backend/tests/unit/dal/Services/PropertyServiceTest.cs (hunk 5 above)
Comment (asanchezr) [reply backend/tests/unit/dal/Services/PropertyServiceTest.cs:31]: ok - will remove inserting the property
